92-AppendixC.Rmd

```diff
@@ -0,0 +1,312 @@
+
+## Importing data from APIs into R
+
+In addition to working with various file formats, we may also need to retrieve data through Application Programming Interfaces (APIs). APIs provide a structured way to access data hosted on external servers and import it directly into R for analysis.
+
+To access this data, you need to understand how to construct API requests. It's crucial to review the documentation of the specific API you intend to use, as each API may have unique endpoints, parameters, and authentication requirements. Pay attention to:
+
+* Endpoints: These are URLs that point to specific data or services.
+* Parameters: Information you pass to the API to customize your request (e.g., date ranges, filters).
+* Authentication: APIs may require API keys or tokens for access.
+* Rate Limits: APIs may have usage limits, so be aware of any rate limits or quotas.
+
+Typically, we begin by making a GET request to an API endpoint. The {httr2} package allows us to generate and process HTTP requests. We can make the GET request by pointing to the URL that contains the data we would like.
+
+```r
+library(httr2)
+
+api_url <- "https://api.example.com/survey-data"
+response <- GET(api_url)
+```
+
+Once we make the request, we will obtain the data as the `response`. The data often comes in JSON format. We can extract and parse the data using the {jsonlite} package, allowing us to work with it in R. The `fromJSON()` function, shown below, coverts JSON data to an R object.
+
+```r
+survey_data <- fromJSON(content(response, "text"))
+```
+
+Note that these are dummy examples; to obtain data from APIs, it is crucial to review the documentation and understand how to make requests.
+
+R offers several packages that simplify API access by providing ready-to-use functions for popular APIs. These packages are called "wrappers", as they "wrap" the API to make it easier to use. For example, the {tidycensus} package used in this book simplifies access to U.S. Census data, allowing us to retrieve data with R commands instead of writing complex API requests. For example, if we are interested in the population (`B01003_001`) of each census tract in North Carolina from the 2020 ACS, we would use the `get_acs()` function and the code below. Behind the scenes, `get_acs()` is making a GET request from the Census API and the tidycensus functions are converting the response into an R-friendly format.
+
+```r
+library(tidycensus)
+
+census_data <- get_acs(geography = "tract", variables = "B01003_001", year = 2020, state = "NC")
+```
+
+To discover if there's an R package that directly interfaces with a specific survey or data source, search for "[survey] R wrapper" or "[data source] R package" online.
```

Comment:
Do we want to add a small section on databases?  It's possible that people at larger companies may have databases.  Or do we consider that out of scope?